server.go

```diff
@@ -2484,12 +2460,20 @@ func (s *server) BanPeer(sp *serverPeer) {
 	if cfg.DisableBanning || sp.isWhitelisted {
 		return
 	}
-	sp.Disconnect()
-
-	select {
-	case <-s.quit:
-	case s.banPeers <- sp:
+	host, _, err := net.SplitHostPort(sp.Addr())
+	if err != nil {
+		srvrLog.Debugf("can't split ban peer %s %v", sp.Addr(), err)
+		return
```

Comment:
This changes the semantics.  Previously, it would log this debug and avoid adding it to the banned peers, but it would still disconnect the peer.  That no longer happens.

Reply:
This is still wrong, but I fixed it in #3309, so going to approve and merge.